src/main/java/org/killbill/billing/entitlement/api/EntitlementApi.java

```diff
@@ -99,6 +99,30 @@ public Entitlement createBaseEntitlement(UUID accountId, PlanPhaseSpecifier spec
     public Entitlement addEntitlement(UUID bundleId, PlanPhaseSpecifier spec, List<PlanPhasePriceOverride> overrides, LocalDate entitlementEffectiveDate,  LocalDate billingEffectiveDate, boolean isMigrated, Iterable<PluginProperty> properties, CallContext context)
             throws EntitlementApiException;
 
+
+    /**
+     * Adds an STANDALONE entitlement to previously created entitlement.
+     * <p/>
+     * The <code>PlanPhaseSpecifier<code/> should refer to a <code>ProductCategory.STANDALONE</code>.
+     * The new entitlement will be bundled using the externalKey that was specified when creating the
+     * base entitlement.
+     *
+     * @param bundleId      the id of the bundle
+     * @param accountId     the account id
+     * @param spec          the product specification for that new entitlement
+     * @param overrides     the price override for each phase and for a specific currency
+     * @param entitlementEffectiveDate the date at which the entitlement should start. if this is null this is assumed now
+     * @param billingEffectiveDate  the date at which the billing for the subscription should start. if this is null this is assumed now
+     * @param isMigrated     whether this subscription comes from a different system (migrated into Kill Bill)
+     * @param properties     plugin specific properties
+     * @param context       the context
+     * @return a new entitlement
+     * @throws EntitlementApiException if the system fail to create the <code>Entitlement</code>
+     */
+    @RequiresPermissions(ENTITLEMENT_CAN_CREATE)
+    public Entitlement addEntitlement(UUID bundleId, UUID accountId, PlanPhaseSpecifier spec, List<PlanPhasePriceOverride> overrides, LocalDate entitlementEffectiveDate,  LocalDate billingEffectiveDate, boolean isMigrated, Iterable<PluginProperty> properties, CallContext context)
```

Comment:
We went back and forth on this with @sbrossie and we would like to suggest another approach:

1. Rename `createBaseEntitlement` to `createBundleWithEntitlement`. The idea is that this call should support creating both a `STANDALONE` and a `BASE` entitlement on a new bundle (the Javadoc actually already says so)
2. Keep `addEntitlement` as-is and make sure it supports adding both types to an existing bundle

Note also that you don't need to pass `UUID accountId` explicitly, even if you need it in the implementation. The `CallContext` already contains the `accountId`.